typed-racket-lib/typed-racket/private/type-contract.rkt

```diff
@@ -464,6 +464,13 @@
             (apply or/sc (append other-scs (map t->sc (nbits->base-types nbits)))))]
        [(? Union? t)
         (match (normalize-type t)
+          [(HashTableTop:)
+           ;; NOTE: this is a special case to make `HashTableTop` produce a flat contract.
```

Comment:
Strictly better for the type HashTableTop, yes.

I don't think it's better for something that could be an `(ImmutableHashTable Any Any)`, but then again it seems like the contract for the `Any` type tends to get in people's way.